common/src/main/java/org/apache/drill/common/expression/SchemaPath.java

```diff
@@ -54,15 +50,16 @@ public static SchemaPath getSimplePath(String name) {
   }
 
   public static SchemaPath getCompoundPath(String... strings) {
-    List<String> paths = Arrays.asList(strings);
-    Collections.reverse(paths);
     NameSegment s = null;
-    for (String p : paths) {
-      s = new NameSegment(p, s);
+    final int length = strings.length;
+    // loop through strings in reverse order
+    for (int i = 0; i < length; i++) {
```

Comment:
Could you change this to (i=length-1; i >= 0; i--) such that you don't have to the extra subtraction below.